drivers/cc1xxx_common/gnrc_netif_cc1xxx.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2016-17 Freie Universität Berlin
+ *               2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc_netif
+ * @{
+ *
+ * @file
+ *
+ * @author      Martine Lenders <m.lenders@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * Adapted from @ref gnrc_xbee.c with only minor modifications. So all credit
+ * goes to Martine Lenders <m.lenders@fu-berlin.de> and to
+ * Hauke Petersen <hauke.petersen@fu-berlin.de>.
+ *
+ * @}
+ */
+
+#include "assert.h"
+#include "net/gnrc.h"
+#include "cc1xxx_common.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define BCAST  (GNRC_NETIF_HDR_FLAGS_BROADCAST | GNRC_NETIF_HDR_FLAGS_MULTICAST)
+
+static gnrc_pktsnip_t *cc1xxx_adpt_recv(gnrc_netif_t *netif)
+{
+    netdev_t *dev = netif->dev;
+    cc1xxx_rx_info_t rx_info;
+    int pktlen;
+    gnrc_pktsnip_t *payload;
+    gnrc_pktsnip_t *hdr;
+    gnrc_netif_hdr_t *netif_hdr;
+    cc1xxx_l2hdr_t l2hdr;
+
+    assert(dev);
+
+    /* see how much data there is to process */
+    pktlen = dev->driver->recv(dev, NULL, 0, &rx_info);
+    if (pktlen <= 0) {
+        DEBUG("[cc1xxx-gnrc] recv: no data available to process\n");
+        return NULL;
+    }
+
+    /* allocate space for the packet in the pktbuf */
+    payload = gnrc_pktbuf_add(NULL, NULL, pktlen, CC1XXX_DEFAULT_PROTOCOL);
+    if (payload == NULL) {
+        DEBUG("[cc1xxx-gnrc] recv: unable to allocate space in the pktbuf\n");
+        /* tell the driver to drop the packet */
+        dev->driver->recv(dev, NULL, 1, NULL);
```

Comment:
Nitpicky: magic number. I propose to use `pktlen` here. This might even be somewhat of an optimization, since `pktlen` should still be in the corresponding register from the `gnrc_pktbuf_add()` call.